project/src/main/webapp/directory-page/directory-page.component.js

```diff
@@ -15,11 +15,64 @@
 // use modern JavaScript (ES5)
 "use strict"
 
-angular.module('directoryPage').component('directoryPage', {
-  templateUrl: '/directory-page/directory-page.template.html', 
-  controller: function directoryPageController() {
-    // TODO: modify this code so it works within the component, and interacts
-    // with the template rather than as a seperate JS function.
-      addBriefElectionCandidateInformation()
+angular.module('directoryPage')
+    .component('directoryPage', {
+      templateUrl: '/directory-page/directory-page.template.html', 
+      controller: function directoryPageController($scope) {
+        showOrHideStateFilter($scope);
+        getElectionCandidateInformation($scope);
+        $scope.filterOnState = function() {
+          getElectionCandidateInformation($scope);
+        }
+        $scope.states =
+            ['--', 'AL', 'AK', 'AR', 'AZ', 'CA', 'CO', 'CT', 'DE', 'FL', 'GA',
```

Comment:
Why the --?

Reply:
This is for when the user decides to go back to viewing all elections. The dropdown box doesn't automatically provide a no-selection option after you selected one.